### PR TITLE
[Feature] MongoDB Atlas: Adds search index commands and tests/examples of metadata filters

### PR DESCRIPTION
**Pull request by caseyclements**
# Description

Adds utility functions to create/update/drop search indexes without having to go through the Atlas UI.
For technical reasons, this currently is only available for clusters of size >=M10, thus not the free sandbox M0's.

Fixes # (issue)
N/A

## New Package?
No

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [X] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
In addition to local unit/integration tests, we have a CI that we run nightly connecting to Local Atlas deployment.
- [X] Added new unit/integration tests

## Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-mongodb/llama_index/vector_stores/mongodb/index.py
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-mongodb/llama_index/vector_stores/mongodb/index.py:1]: Wait I'm confused, where are these methods used? Shouldn't the be used in the vector store class?

Comment (caseyclements) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-mongodb/llama_index/vector_stores/mongodb/index.py:1]: Hi @logan-markewich . I'm happy to refactor things to expose the searches index operations from the vector store directly. I'll do so in the next week.

**Comment by blink1073** [comment]
Hi @logan-markewich, we've refactored to expose the search index helpers on the `MongoDBAtlasVectorSearch` class.

**Review by logan-markewich — COMMENTED** [review]
Looks good, my only request is, can we update the [README](https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/vector_stores/llama-index-vector-stores-mongodb/README.md) and [notebook](https://github.com/run-llama/llama_index/blob/main/docs/docs/examples/vector_stores/MongoDBAtlasVectorSearch.ipynb) to show the full code flow?

**Comment by caseyclements** [comment]
Hi @logan-markewich, @blink1073  I think we're good to go. You coverage is failing because you'd need a secret or two from us. This is what I get on my side.

```
---------- coverage: platform darwin, python 3.9.20-final-0 ----------
Name                                             Stmts   Miss  Cover
--------------------------------------------------------------------
llama_index/vector_stores/mongodb/__init__.py        2      0   100%
llama_index/vector_stores/mongodb/base.py          138     18    87%
llama_index/vector_stores/mongodb/index.py          56     10    82%
llama_index/vector_stores/mongodb/pipelines.py      47      9    81%
--------------------------------------------------------------------
TOTAL                                              243     37    85%
```

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**